pkg/identity/certificated.go

```diff
@@ -102,17 +102,26 @@ func Certificated(idConfig *config.IdentityConfig, stopChan <-chan struct{}) (er
 					if err := w.AddBytes(outPath, 0644, roleCertPEM); err != nil {
 						return errors.Wrap(err, "unable to save x509 role cert")
 					}
+
+					// always output role cert key file to prevent unexpected key rotation when using external key
+					if id == nil || idConfig.RoleCertKeyFileOutput {
```

Comment:
```suggestion
					if id != nil && idConfig.RoleCertKeyFileOutput {
```

Not this??

Reply:
ONLY allow disable on single SIA pattern (internal key rotation).
`id == nil` =>  NO internal rotation => always `true`

### cases
1. non-copperargos, external rotation, always `true`
2. sidecar pattern (non-copper, issue role cert ONLY), another SIA in copperargos, always `true`
3. copperargos, single SIA, allow `true/false`

Reply:
@WindzCUHK Ah, makes sense. Thank you.

Reply:
Kinda wonder how end user would feel when they set `RoleCertKeyFileOutput` as `false` and still outputs the role key.

Reply:
`RoleCertKeyFileOutput` => `RoleCertKeyFileOutputOnInternalKey`?

Reply:
@WindzCUHK 
Mmm lets just stick with `RoleCertKeyFileOutput`.


> Kinda wonder how end user would feel when they set RoleCertKeyFileOutput as false and still outputs the role key.

nvm